system/jlib/jtrace.cpp

```diff
@@ -427,31 +436,31 @@ class CSpan : public CInterfaceOf<ISpan>
         tracerName.set(nameOfTracer);
     }
 
-    void init()
+    void init(SpanFlags flags)
     {
         bool createLocalId = !isEmptyString(hpccGlobalId);
         if (createLocalId)
             hpccLocalId.set(ln_uid::createUniqueIdString().c_str());
 
-        auto provider = opentelemetry::trace::Provider::GetTracerProvider();
-
-        //what if tracerName is empty?
-        auto tracer = provider->GetTracer(tracerName.get());
+        //we don't always want to create trace/span IDs
+        if (hasMask(flags, SpanFlags::EnsureTraceId) || queryTraceManager().alwaysCreateTraceIds())
```

Comment:
This doesn't seem quite right.  You still want to create a span if a trace-id was provided by the caller.  It is only omitted if it is not provided and ids are not created by default.